### PR TITLE
Conditional deploy docs.

### PR DESCRIPTION
**Pull request by kellertuer**
Let's first test that this does currently not deploy (trying to solve #148).

We should maybe at some point also delete old PRs preview docs.

### CODE REVIEW

**Comment by kellertuer** [comment]
Check 1 was successful, it does not push when the label is not present.

**Comment by kellertuer** [comment]
Reopening with the label and the docs are pushed. Great!

File: docs/make.jl
Comment (kellertuer) [inline docs/make.jl:98]: One could even speed this up and avoid deploying by adding ` should_push_preview() &&` upfront this line; what do you think?

**Comment by sethaxen** [comment]
I'm not a big fan of the Github Actions-specific logic being in `make.jl`. I think it would be cleaner for `should_push_preview` to just read an environment variable, something like `DOCUMENTER_PUSH_PREVIEW`, defaulting to false if the variable is undefined or not `TRUE` or something like that. (I always forget how to write booleans in shell script). Then update `documenter.yml` to set that environment variable if the label is present.

File: docs/make.jl
Comment (sethaxen) [reply docs/make.jl:98]: If it's not deploying, I doubt calling `deploydocs` with `push_preview=false` takes much time at all.

**Comment by kellertuer** [comment]
It is not Github-Actions intervowen, it actually just reads Github Environment variables (but not a `DOCUMENTER_PUSH_PREVIEW`), namely the labels that are attached to a PR. If its not a PR, I think the `event_path` is not given (since we are on master) and docs are pushed anyways.

File: docs/make.jl
Comment (kellertuer) [reply docs/make.jl]: You're right. Anyways with your other remark and now an approach to a pure GitHub Actions approach this point is not necessary anymore :)

**Comment by kellertuer** [comment]
Now this PR contains a solution purely based on GitHub Actions:
Documenter now contains two actions:
* Documenter, which only runs if the `preview docs` label is present
* a second action (Documentation deployment note), that only runs if the label is _not_ present

this second action has two purposes: (1) it informs a user how to turn docs-preview on if they are off and (2) the action is not empty when the label is not present, because an empty action would fail

Despite that there is only again something with formatter – even updating it locally here and running it still reports errors on the CI; seems like formatter changes its defaults a lot currently..., which is a little sad.

File: docs/Project.toml
Comment (sethaxen) [inline docs/Project.toml:5]: Is JSON still needed?
```suggestion
JSON = "682c06a0-de6a-54ab-a142-c8b1cf79cde6"
```

File: docs/make.jl
Comment (sethaxen) [inline docs/make.jl:1]: Likewise, do we still need this import?
```suggestion
using Manifolds, ManifoldsBase, Documenter
```

**Review by sethaxen — APPROVED** [review]
Wow this ended up being really clean. LGTM, once the JSON dependency is clarified or removed. I'll let you address and merge when ready.

File: docs/make.jl
Comment (kellertuer) [reply docs/make.jl]: Thanks for noticing, no we don't.

**Comment by kellertuer** [comment]
Thanks for noticing the JSON dependency, that is of course not necessary;
and well – I saw the label conditional for github actions and you were right, decoupling it as it is now is of course much cleaner.

**Pull request merged**
